scripts/audit_branch_protection.sh

```
@@ -0,0 +1,164 @@
+# Compose the final ``{"rulesets":[...]}`` document from the per-ID
+# files in a deterministic order (sort by numeric id) so diffs stay
+# stable across runs.
+{
+  printf '{"rulesets":['
+  first=1
+  # Skipping ``-r`` here is intentional -- ``id`` is a bare integer.
+  while IFS= read -r id; do
+    [ -z "$id" ] && continue
+    if [ "$first" -eq 1 ]; then first=0; else printf ','; fi
+    cat "${RULESETS_DIR}/${id}.json"
+  done < <(printf '%s\n' $IDS | sort -n)
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)